src/ucs/datastruct/piecewise_func.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <ucs/datastruct/piecewise_func.h>
+#include <ucs/debug/memtrack_int.h>
+#include <ucs/debug/assert.h>
+#include <limits.h>
+#include <stdint.h>
+
+static ucs_piecewise_segment_t *ucs_piecewise_segment_alloc()
+{
+    return ucs_malloc(sizeof(ucs_piecewise_segment_t), "piecewise segment");
+}
```

Comment:
i'd just use malloc directly instead of this wrapper

Reply:
Done.